src/main/java/one/util/streamex/MoreCollectors.java

```diff
@@ -168,6 +168,209 @@ private MoreCollectors() {
+     * the provided {@code valueMapper} function to {@code Map.Entry} values.
+     * @throws IllegalStateException if this stream contains duplicate keys
+     *                               (according to {@link Object#equals(Object)})
+     *
+     * @see Collectors#toMap(Function, Function)
+     */
+    public static <K, V, VV> Collector<Entry<K, V>, ?, Map<K, VV>> entriesToMap(Function<V, VV> valueMapper) {
+        return Collectors.toMap(Entry::getKey, entry -> valueMapper.apply(entry.getValue()));
+    }
+
+    /**
+     * Returns a {@code Collector} that accumulates input elements into a {@code Map}
+     * whose keys are taken from {@code Map.Entry} and values are the result
+     * of applying the provided {@code valueMapper} function to {@code Map.Entry} values
+     * and combining them using the provided {@code combiner} function.
+     *
+     * @param <K>         the {@link Comparable} type of then map keys
+     * @param <V>         the type of the map values
+     * @param <VV>        the output type of the value mapping function
+     * @param valueMapper a mapping function to produce values
+     * @param combiner    a merge function, used to resolve collisions between
+     *                    values associated with the same key, as supplied
+     *                    to {@link Map#merge(Object, Object, BiFunction)}
+     * @return {@code Collector} which collects elements into a {@code Map}
+     * whose keys are taken from {@code Map.Entry} and values are the result
+     * of applying the provided {@code valueMapper} function and combining
+     * them using the provided {@code combiner} function.
+     *
+     * @see Collectors#toMap(Function, Function, BinaryOperator)
+     */
+    public static <K, V, VV> Collector<Entry<K, V>, ?, Map<K, VV>> entriesToMap(
+            Function<V, VV> valueMapper, BinaryOperator<VV> combiner) {
+        Objects.requireNonNull(valueMapper);
+        return Collectors.toMap(Entry::getKey, entry -> valueMapper.apply(entry.getValue()), combiner);
+    }
+
+    /**
+     * Returns a {@code Collector} that accumulates elements into
+     * a result {@code Map} defined by {@code mapSupplier} function
+     * whose keys and values are taken from {@code Map.Entry}.
+     *
+     * @param <K> the {@link Comparable} type of then map keys
+     * @param <V> the type of the map values
+     * @param <M> the type of the resulting {@code Map}
+     * @return {@code Collector} which collects elements into a {@code Map}
+     * defined by {@code mapSupplier} function
+     * whose keys and values are taken from {@code Map.Entry}
+     * @throws IllegalStateException if this stream contains duplicate keys
+     *                               (according to {@link Object#equals(Object)})
+     * @see Collectors#toMap(Function, Function, BinaryOperator, Supplier)
+     */
+    public static <K, V, M extends Map<K, V>> Collector<Entry<K, V>, ?, M> entriesToCustomMap(
```

Comment:
Please document NPE in case of `mapSupplier` is null (and ensure it's actually thrown). Same for other added methods.